### ISSUE TITLE
Invalid BED format from duplicate columns

### ISSUE THREAD

**Issue by michaelGuo1204** [gao-lab/GLUE#69]
Hi, I have the same problem as  #66  from viet and I found that `rna.var` dataframe of  `Chen-2019-RNA.h5ad`  has duplicated columns after annotation via `scglue.data.get_gene_annotation`. Remove these duplicates and the whole pipeline just works. 

 ```
genes     chrom  chromStart   chromEnd ......         chrom  chromStart   chromEnd                       
0610005C13Rik   chr7    45567793   45575327 ...... chr7    45567793   45575327
0610009B22Rik  chr11    51685385   51688874 ...... chr11    51685385   51688874
0610009E02Rik   chr2    26445695   26459390 ...... chr2    26445695   26459390
```
I also noticed that the unit test of  `scglue.data.get_gene_annotation` does not capture such duplicating behaviour. So I checked the function definition and found `pd.concat` operation within that function might lead to repeat columns.

```python
merge_df = pd.concat([
        pd.DataFrame(gtf.to_bed(name=gtf_by)),
        pd.DataFrame(gtf).drop(columns=genomics.Gtf.COLUMNS)  # Only use the splitted attributes
    ], axis=1).set_index(gtf_by).reindex(var_by).set_index(adata.var.index)
adata.var = pd.concat([adata.var, merge_df], axis=1)
```

Add a `drop_duplicates()` then it works.

P.S. I use the recommended packages so it should not be the package version's issue

**Comment by Jeff1995** [comment]
Hi @michaelGuo1204. Thanks for the detailed report!

The tutorial was meant to be run on a `var`-cleared version of the `Chen-2019-RNA.h5ad` file listed on the tutorial page:

http://download.gao-lab.org/GLUE/tutorial/Chen-2019-RNA.h5ad

But due to the exact reason you described, would throw this error when run on the `var`-populated version listed on the example datasets page (yes these files are different...):

http://download.gao-lab.org/GLUE/dataset/Chen-2019-RNA.h5ad

Not properly handling duplicate columns is indeed very confusing. I have pushed a fix (679989bcf551d3b53c0d39b6db98db520797eb16) by replacing the `concat` with `assign`, which will overwriting the existing columns when there are duplications. The fix will be included in the next release.

**Comment by michaelGuo1204** [comment]
Appreciate it!
